icu4c/source/common/putilimp.h

```diff
@@ -101,7 +101,7 @@ typedef size_t uintptr_t;
 #if U_PLATFORM_HAS_WINUWP_API == 0
 #   define U_TZSET _tzset
 #endif
-#elif U_PLATFORM == U_PF_OS400
+#elif U_PLATFORM == U_PF_OS400 || defined(__HAIKU__)
```

Comment:
Should be able to use U_PLATFORM == U_HAIKU ?

Reply:
Let me check this today, will get back to this, thanks!

Reply:
Should be fixed.

Reply:
Hi @srl295 you suggested using U_PLATFORM == U_HAIKU, but there is no U_HAIKU in unicode/platform.h or anywhere else.

Should we go back to testing if `__HAIKU__` is defined?

Should we define U_HAIKU? If so, what would be a good numeric value? Is it similar to something else and could it benefit from numeric-value range checks for supporting similar platforms?
Maybe figuring out a good numeric value and defining U_HAIKU is better in a follow-up PR.

Reply:
For the build failure:
https://github.com/unicode-org/icu/actions/runs/10997790140/job/30534185446?pr=2463
```
In file included from uvectr64.cpp:12:
./putilimp.h:93:21: error: 'U_HAIKU' is not defined, evaluates to 0 [-Werror,-Wundef]
*** Failed compilation command follows: ----------------------------------------------------------
#elif U_PLATFORM == U_HAIKU
                    ^
./putilimp.h:108:21: error: 'U_HAIKU' is not defined, evaluates to 0 [-Werror,-Wundef]
#elif U_PLATFORM == U_HAIKU
                    ^
2 errors generated.
```

Reply:
Thanks for rechecking, almost forgot about this one :) Will check against my current 75.1 work/patch, and maybe to a source (git) checkout to see if anything changed.

Reply:
A quick check resulted into the next, not sure if it's acceptable?

```diff
diff --git a/source/common/putilimp.h b/source/common/putilimp.h
index 29dcf0e..025dfd9 100644
--- a/source/common/putilimp.h
+++ b/source/common/putilimp.h
@@ -90,7 +90,7 @@ typedef size_t uintptr_t;
 #   define U_NL_LANGINFO_CODESET -1
 #elif U_PLATFORM == U_PF_OS400
    /* not defined */
-#elif U_PLATFORM == U_HAIKU
+#elif U_PLATFORM == U_PF_HAIKU
    /* not defined */
 #else
 #   define U_NL_LANGINFO_CODESET CODESET
@@ -105,7 +105,7 @@ typedef size_t uintptr_t;
 #endif
 #elif U_PLATFORM == U_PF_OS400
    /* not defined */
-#elif U_PLATFORM == U_HAIKU
+#elif U_PLATFORM == U_PF_HAIKU
    /* not defined */
 #else
 #   define U_TZSET tzset
@@ -145,7 +145,7 @@ typedef size_t uintptr_t;
 #endif
 #elif U_PLATFORM == U_PF_OS400
    /* not defined */
-#elif U_PLATFORM == U_HAIKU
+#elif U_PLATFORM == U_PF_HAIKU
    /* not defined, (well it is but a loop back to icu) */
 #else
 #   define U_TZNAME tzname
diff --git a/source/common/unicode/platform.h b/source/common/unicode/platform.h
index 7aca76c..5e79335 100644
--- a/source/common/unicode/platform.h
+++ b/source/common/unicode/platform.h
@@ -133,6 +133,8 @@
 /** Android is based on Linux. @internal */
 #define U_PF_ANDROID 4050
 /** Fuchsia is a POSIX-ish platform. @internal */
+#define U_PF_HAIKU 4080
+/** Haiku is a POSIX-ish platform. @internal */
 #define U_PF_FUCHSIA 4100
 /* Maximum value for Linux-based platform is 4499 */
 /**
@@ -200,6 +202,8 @@
 #   define U_PLATFORM U_PF_OS390
 #elif defined(__OS400__) || defined(__TOS_OS400__)
 #   define U_PLATFORM U_PF_OS400
+#elif defined(__HAIKU__)
+#   define U_PLATFORM U_PF_HAIKU
 #elif defined(__EMSCRIPTEN__)
 #   define U_PLATFORM U_PF_EMSCRIPTEN
 #else
```

Still need to check this with the haikuports team also.

Reply:
Seems like the comment for Fuchsia is misplaced in the patch... Needs to moved back after the Haiku lines that are being added...

Reply:
Right, this should look like
```
+/** Haiku is a POSIX-ish platform. @internal */
+#define U_PF_HAIKU 4080
 /** Fuchsia is a POSIX-ish platform. @internal */
 #define U_PF_FUCHSIA 4100
```

Reply:
My bad, it's getting late(r) and wanted to present something. Will fix tomorrow. :)

Reply:
Will give this another look in the morning with a clear head and some coffee, I think a small change for this in configure.ac could be added.

Reply:
That's what for reviews are... Different people look for different things when reviewing... We all make mistakes even after being developers for several decades.

Reply:
It's good if we don't have to rely on configure/autoconf/autotools, because not everyone is building ICU with that build system.

Also, autoconf and makefiles are really clunky, and various users are clamoring for moving to a more modern build system (more recently working on using Meson).

Reply:
Some people are still using quite old operating systems and old hardware as in a lot of countries people buy new hardware once in 9~10 years as it's very expensive. As such I've had many issues with more modern alternatives like CMake. Even though autoconf/configure and automake/Makefiles are quite a mess, they still run better than some of the modern alternatives.